src/coreclr/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -532,6 +544,142 @@ public static int GetCurrentProcessorId()
             return currentProcessorIdCache >> ProcessorIdCacheShift;
         }
 
+        // do a fast check and record in a readonly static so that it could become a JIT constant
+        internal static readonly bool IsCoreIdReallyFast = SimpleCoreIdSpeedCheck();
+
+        // If GetCurrentProcessorNumber takes any nontrivial time (compared to TLS access), return false.
+        // Check more than once - to make sure it was not because TLS was delayed by GC or a context switch.
+        private static bool SimpleCoreIdSpeedCheck()
+        {
+            // warm up the code paths.
+            int id = UninlinedThreadStatic() | GetCurrentProcessorNumber();
+            long _05usec = Stopwatch.Frequency / 1000000;
```

Comment:
Nit: Local names should not start with `_`

Reply:
C# does not let me have a name starting with a number :-\   I guess this is too cute. I will make these start with `t`.

---
In reply to: [352951996](https://github.com/dotnet/runtime/pull/467#discussion_r352951996) [](ancestors = 352951996)

Reply:
you could call it `halfMicrosec`?